### ISSUE TITLE
[Feature Request] Implement Iterator interface in Recursive_ArrayAccess class

### ISSUE THREAD

**Issue by ste93cry** [ericmann/wp-session-manager#26]
Hello,
you're plugin is great, however a missing feature makes it difficult to use it. The `Recursive_ArrayAccess` class doesn't implement support for iterators, this means that I can't loop the container object to get all data stored as an array without using the `toArray` method. I spent hours trying to figure out why I couldn't use the `foreach` because of this, so my proposal is to implement the `Iterator` interface. It would be cool also to release a new version of the plugin with the compatibility flat set up to 4.0. Thank you and keep up your good work!

**Comment by wielsma** [comment]
Good suggestion @ste93cry, I was having the same problem and spend about an hour to figure out why I couldn't loop through all stored elements.

I found that simply decoding the JSON output from the $wp_session instance was the quickest way to get what I need.

For instance:
`$wp_session_array = json_decode( $wp_session->json_out() );`

Cheers

- Jelle

**Comment by ste93cry** [comment]
Looking at the source code here on GitHub I noticed that `Recursive_ArrayAccess` implements the `Iterator` interface since April 2014. At the time I opened this issue, I used the plugin by downloading it from the WordPress plugins directory and the version published there is 1.1.2, which it's too old and did not support the iterator. Checking right now, the [plugin page](https://wordpress.org/plugins/wp-session-manager/) on WordPress.com has never been updated in 2 years.
